src/wallet/wallet.h

```diff
@@ -715,6 +715,35 @@ class COutput
     std::string ToString() const;
 };
 
+/**
+ * A strategy to use for managing privacy when constructing a transaction.
+ */
+enum class PrivacyPolicy {
+    FullPrivacy,
+    AllowRevealedAmounts,
+    AllowRevealedRecipients,
+    AllowRevealedSenders,
+    AllowFullyTransparent,
+    AllowLinkingAccountAddresses,
+    NoPrivacy,
```

Comment:
I'm not sure that the name `NoPrivacy` is a good idea; could imply that we're relinquishing all privacy even when not necessary, when the actual behavior is more nuanced than this. Perhaps this should be weakened to `NoGuaranteedPrivacy` or something else?

Reply:
I wanted this option to sound scary precisely so that users _don't_ just enable it all the time, because "no privacy" _is_ one of the permitted resulting transaction states (even though the wallet will always try to do better). The policy itself is (I think) well-documented in the help text:
```
- "NoPrivacy": Allow the transaction to reveal any information necessary to create it.
  Implies "AllowFullyTransparent" and "AllowLinkingAccountAddresses".
```
If you can think of a scarier name, or one that is closer to the intent without sounding not-scary, I'm all ears!

Reply:
`AllowFullyTransparentAndLinkingAccountAddresses`?